modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientHandlerModule.java

```diff
@@ -334,6 +340,15 @@ protected void initChannel(Channel ch) {
                 return;
             }
 
+            if (bindFut.cause() instanceof UnresolvedAddressException) {
+                result.completeExceptionally(
+                        new IgniteException(
+                                INTERNAL_ERR,
+                                "Failed to start thin connector endpoint, address \"" + host + "\" is not found"
+                        )
+                );
```

Comment:
I guess you are missing a `return` here. But I would actually prefer to refactor this code to look like this:
```
if (bind.isSuccess()) {
    ...
} else if (bindFut.cause() instanceof BindException) {
    ...
} else if (bindFut.cause() instanceof UnresolvedAddressException) {
    ...
} else {
   ...
}